### PR TITLE
Various Updates to StatChecker

### PR DESCRIPTION
**Pull request by chejennifer**
- For DataHole counter, show list of all dates in the series
- For percentDifference values, multiply by 100 and truncate to 2 digit decimal
- Remove StatsCheck_MaxPercentFluctuationGreaterThan50 counter
- Limit number of StatValidationResult entries to covering 10 of each counter

Additional fix that is not related to StatChecker:
- Generated table mcf node files should have mcf extension instead of csv extension

### CODE REVIEW

File: util/src/main/java/org/datacommons/util/StatChecker.java [hunk 1]
```diff
@@ -126,8 +140,19 @@ public synchronized void check() {
...
+        // entries for.
+        boolean shouldAddEntry = false;
+        for (StatValidationEntry entry : resBuilder.getValidationCountersList()) {
+          String counterKey = entry.getCounterKey();
+          if (countersRemaining.containsKey(counterKey)) {
+            shouldAddEntry = true;
+            countersRemaining.compute(counterKey, (k, v) -> v != null ? v - 1 : 0);
+            if (countersRemaining.get(counterKey) < 1) countersRemaining.remove(counterKey);
+          }
+          if (shouldAddEntry) logCtx.addStatsCheckSummaryEntry(resBuilder.build());
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/StatChecker.java:154]: Should this be outside the for loop?

File: util/src/main/java/org/datacommons/util/StatChecker.java [hunk 2]
```diff
@@ -390,8 +415,12 @@ protected static void checkDates(
           StatValidationEntry.Builder dataHoleCounter = StatValidationEntry.newBuilder();
           String dataHoleCounterKey = "StatsCheck_Data_Holes";
           dataHoleCounter.setCounterKey(dataHoleCounterKey);
+          List<String> dateList = new ArrayList<>();
+          for (DataPoint dp : timeSeries) {
+            dateList.add(dp.getDate());
+          }
           dataHoleCounter.setAdditionalDetails(
-              "Data hole found between the dates: " + prev.toString() + " and " + dt.toString());
+              "Possible data hole found. Dates in this series: " + String.join(",", dateList));
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/StatChecker.java:423]: ", " for readability?

File: tool/src/main/java/org/datacommons/tool/WriterPair.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public void close() throws IOException {
   private BufferedWriter newWriter(Args.OutputFileType type) throws IOException {
     String filePath = args.outputFiles.get(type).toString();
     if (csvFile != null) {
-      String fileSuffix = FilenameUtils.removeExtension(csvFile.getName()) + ".csv";
+      String fileSuffix = FilenameUtils.removeExtension(csvFile.getName()) + ".mcf";
```
Comment (pradh) [inline tool/src/main/java/org/datacommons/tool/WriterPair.java:52]: When I made this change I think I also broke genmcf test (https://github.com/datacommonsorg/import/blob/master/tool/src/test/java/org/datacommons/tool/GenMcfTest.java#L70-L76). I'll fix that after this goes in...

**Review by pradh — APPROVED** [review]
Thanks for the fixes!

File: util/src/main/java/org/datacommons/util/StatChecker.java (hunk 1 above)
Comment (chejennifer) [reply util/src/main/java/org/datacommons/util/StatChecker.java]: good catch, done

File: util/src/main/java/org/datacommons/util/StatChecker.java (hunk 2 above)
Comment (chejennifer) [reply util/src/main/java/org/datacommons/util/StatChecker.java]: done

File: tool/src/main/java/org/datacommons/tool/WriterPair.java (hunk 3 above)
Comment (chejennifer) [reply tool/src/main/java/org/datacommons/tool/WriterPair.java:52]: ok, sounds good

**Comment by chejennifer** [comment]
Thanks for the review!

**Pull request merged**